src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedEventTest.kt

```diff
@@ -0,0 +1,142 @@
+    assertThat(runCatching { getInductionSchedule(prisonNumber) }.getOrNull()).isNull()
+
+    val earliestDateTime = OffsetDateTime.now()
+
+    val sqsMessage = prisonerReceivedSqsMessage(prisonNumber)
+
+    // When
+    sendDomainEvent(sqsMessage)
+
+    // Then
+    // wait until the queue is drained / message is processed
+    await untilCallTo {
+      domainEventQueueClient.countMessagesOnQueue(domainEventQueue.queueUrl).get()
+    } matches { it == 0 }
+
+    // test induction schedule was created
+    val inductionSchedule = getInductionSchedule(prisonNumber)
+    assertThat(inductionSchedule)
+      .wasCreatedAfter(earliestDateTime)
+      .wasUpdatedAfter(earliestDateTime)
+      .wasCreatedBy("system")
+      .wasCreatedByDisplayName("system")
+      .wasUpdatedBy("system")
+      .wasUpdatedByDisplayName("system")
+      .wasScheduleCalculationRule(InductionScheduleCalculationRuleResponse.NEW_PRISON_ADMISSION)
+      .wasStatus(InductionScheduleStatusResponse.SCHEDULED)
+
+    // test induction schedule history is created
+    val inductionScheduleHistories = getInductionScheduleHistory("A6099EA")
+    assertThat(inductionScheduleHistories.inductionSchedules).size().isEqualTo(1)
+    assertThat(inductionScheduleHistories.inductionSchedules[0])
+      .wasCreatedAfter(earliestDateTime)
+      .wasUpdatedAfter(earliestDateTime)
+      .wasCreatedBy("system")
+      .wasCreatedByDisplayName("system")
+      .wasUpdatedBy("system")
+      .wasUpdatedByDisplayName("system")
+      .wasScheduleCalculationRule(InductionScheduleCalculationRuleResponse.NEW_PRISON_ADMISSION)
+      .wasStatus(InductionScheduleStatusResponse.SCHEDULED)
+      .wasVersion(1)
+
+    // test that outbound event is also created:
+    val inductionScheduleEvent = inductionScheduleEventQueue.receiveEvent(QueueType.INDUCTION)
+    assertThat(inductionScheduleEvent.personReference.identifiers[0].value).isEqualTo("A6099EA")
+    assertThat(inductionScheduleEvent.detailUrl).isEqualTo("http://localhost:8080/inductions/A6099EA/induction-schedule")
+  }
+
+  @Test
+  fun `should update the Review Schedule and send outbound message given 'prisoner received' event for prisoner that already has an Induction and initial Review Schedule`() {
+    // Given
+    // an induction schedule, induction and action plan are created. This will have created the initial Review Schedule
+    val prisonNumber = randomValidPrisonNumber()
+    createInductionSchedule(
+      prisonNumber = prisonNumber,
+      status = InductionScheduleStatus.COMPLETE,
+    )
+    createInduction(prisonNumber, aValidCreateInductionRequestForPrisonerNotLookingToWork())
+    createActionPlan(prisonNumber)
+
+    // The above calls set the data up but they will also generate events so clear these out before starting the test.
+    // Before clearing the queues though we need to wait until the first "plp.review-schedule.updated" event on the REVIEW queue is received.
+    await untilCallTo {
```

Comment:
This is the crux of the problems I was having.
The `plp.review-schedule.updated` is actually sent twice in this test - once in the `// Given` section as part of the creation of the Induction and Action Plan (this was a recent change, to create the initial Review Schedule on creation of the Induction and 1st Goal; which probably explains why we've not seen this as a problem until now)

Anyway, even though the impl doesnt use an `AsyncEvent....` type class so in theory is not async, the internals of the AWS SNS & SQS etc classes use Completable Futures, so is async!

Because it's async, when we hit the `clearQueues` method we cant guarantee that the SNS message has been sent and mapped (by localstack) to the SQS queue. That is what was happening for me (likely because my machine is slow today and/or localstack is on its knees)

So .... by the time the final assertion block on line 121 was hit:
```
val reviewScheduleEvent = inductionScheduleEventQueue.receiveEvent(QueueType.REVIEW)
```
most times there were 2 messages on the queue!

The fix is to add the `await` until there is at least one message on the queue before clearing the queues.

Phew! Thats a long explanation !